### PR TITLE
Adds roles and permissions

### PR DESCRIPTION
**Pull request by dborsatto**

### CODE REVIEW

File: tests/E2E/RoleTest.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        $manager = $this->getReadWriteSpaceManager();
+        $roles = $manager->getRoles();
+
+        $this->assertInstanceOf(Role::class, $roles[0]);
+
+        $query = (new Query)
+            ->setLimit(1);
+        $roles = $manager->getRoles($query);
+        $this->assertInstanceOf(Role::class, $roles[0]);
+        $this->assertCount(1, $roles);
```
Comment (mariobodemann) [inline tests/E2E/RoleTest.php]: Should you also assert on whether the content of the roles array is the same, not just the count (aka can you check if the values set on the first entry are the ones you would expect?)

File: tests/E2E/RoleTest.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        $this->assertInstanceOf(AndConstraint::class, $policies[0]->getConstraint());
+        $constraint = $policies[0]->getConstraint()->getChildren()[0];
+        $this->assertInstanceOf(EqualityConstraint::class, $constraint);
+        $this->assertEquals('sys.type', $constraint->getDoc());
+        $this->assertEquals('Entry', $constraint->getValue());
+
+        $permissions = $role->getPermissions();
+        $this->assertEquals('all', $permissions->getContentDelivery());
+        $this->assertEquals(['read'], $permissions->getContentModel());
+        $this->assertEquals([], $permissions->getSettings());
```
Comment (mariobodemann) [inline tests/E2E/RoleTest.php]: 💕

File: tests/E2E/RoleTest.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        $policy->setConstraint($constraint);
+
+        $manager->create($role);
+
+        $this->assertNotNull($role->getSystemProperties()->getId());
+        $this->assertEquals(0, $role->getSystemProperties()->getVersion());
+
+        // The ResourceBuilder actually recreates from scratch all attributes,
+        // using the result of the CMA response.
+        // This is to make sure that all attributes were correctly recreated.
```
Comment (mariobodemann) [inline tests/E2E/RoleTest.php]: 😍

**Comment by dborsatto** [comment]
Of course, I'll add the missing tests 🙂

**Comment by dborsatto** [comment]
@mariobodemann I've updated the tests (squashed 2 commits into 1), what do you say?

**Pull request merged**